cw-dex/tests/astroport_tests.rs

```diff
@@ -0,0 +1,306 @@
+#[test_case(PairType::Xyk{},vec![("uluna",1_000_000), ("astro", 1_000_000)], Uint128::new(100_000_000); "swap_and_simulate_swap: basic pool, high slippage")]
+#[test_case(PairType::Xyk{},vec![("uluna",68_582_147), ("astro", 3_467_256)], Uint128::new(1_000_000); "swap_and_simulate_swap: basic pool, random prices")]
+#[test_case(PairType::Stable { },vec![("uluna",1_000_000), ("astro", 1_000_000)], Uint128::new(1_000_000); "swap_and_simulate_swap: stable swap pool")]
+#[test_case(PairType::Stable { },vec![("uluna",1_000_000), ("astro", 1_000_000)], Uint128::new(100_000_000); "swap_and_simulate_swap: stable swap pool, high slippage")]
+#[test_case(PairType::Stable { },vec![("uluna",68_582_147), ("astro", 3_467_256)], Uint128::new(1_000_000); "swap_and_simulate_swap: stable swap pool, random prices")]
+fn test_swap_and_simulate_swap(
+    pool_type: PairType,
+    initial_liquidity: Vec<(&str, u64)>,
+    amount: Uint128,
+) {
+    let (runner, accs, _lp_token_addr, _pair_addr, contract_addr, asset_list) =
+        setup_pool_and_contract(pool_type, initial_liquidity).unwrap();
+
+    let admin = &accs[0];
+    let wasm = Wasm::new(&runner);
+
+    let offer_info = &asset_list.to_vec()[0].info;
+    let ask_info = &asset_list.to_vec()[1].info;
+
+    // Simulate swap
+    let offer = Asset {
+        info: offer_info.clone(),
+        amount,
+    };
+    let simulate_query = QueryMsg::SimulateSwap {
+        offer: offer.clone(),
+        ask: ask_info.clone(),
+        sender: None,
+    };
+
+    let expected_out = wasm.query(&contract_addr, &simulate_query).unwrap();
+
+    // Swap
+    let swap_msg = ExecuteMsg::Swap {
+        offer: offer.clone(),
+        ask: ask_info.clone(),
+        min_out: expected_out,
+    };
+    let native_coins = match offer.info {
+        AssetInfoBase::Native(denom) => {
+            vec![Coin {
+                denom,
+                amount: offer.amount,
+            }]
+        }
+        AssetInfoBase::Cw20(cw20_addr) => {
+            // Transfer cw20 tokens to the contract
+            cw20_transfer(
+                &runner,
+                cw20_addr.to_string(),
+                contract_addr.clone(),
+                offer.amount,
+                admin,
+            )
+            .unwrap();
+            vec![]
+        }
+    };
+    runner
+        .execute_cosmos_msgs::<MsgExecuteContractResponse>(
+            &[swap_msg.into_cosmos_msg(contract_addr.clone(), native_coins)],
+            admin,
+        )
+        .unwrap();
+
+    // Query offer and ask balances
+    let offer_balance = query_asset_balance(&runner, offer_info, &contract_addr);
+    let ask_balance = query_asset_balance(&runner, ask_info, &contract_addr);
+
+    // Assert that offer and ask balances are correct
+    assert_eq!(ask_balance, expected_out);
+    assert_eq!(offer_balance, Uint128::zero());
+}
```

Comment:
can't find tests for `simulate_withdraw_liquidity`, `simulate_provide_liquidity` , `query_pool_info`,  `query_lp_token_supply`, `get_pool_liquidity` for `AstroportPool` and `claim_rewards`, `query_pending_rewards` for `AstroportStaking`.

Reply:
Opened a new issue for those